src/border_agent/border_agent.cpp

```diff
@@ -207,6 +207,16 @@ void BorderAgent::SetEnabled(bool aIsEnabled)
     return;
 }
 
+void BorderAgent::SetEpskcEnabled(bool aIsEnabled)
+{
+    VerifyOrExit(IsEpskcEnabled() != aIsEnabled);
+    mIsEpskcEnabled = aIsEnabled;
+    UpdateMeshCopService();
```

Comment:
Do we intend to allow `SetEpskcEnabled()` to be called at any point, especially while e-PSK is actively in use?

Consider this scenario:

* The e-PSK feature is enabled and actively used (e-PSKs are set and the related service is being published).
* While this is happening, `SetEpskcEnabled(false)` is called.

Will the system correctly stop using e-PSK and unpublish the associated service?

While I see that this PR adds code to `PublishEpskcService()` to check `IsEpskcEnabled()` before publishing, I'm not sure how we would handle unpublishing the service in the scenario described above.

Reply:
My suggestion is that whenever `mIsEpskcEnabled` transitions from `true` to `false`, we call `otBorderAgentClearEphemeralKey()`. 

* If there's no active e-PSK in use, this call will have no effect.
* If there is an active e-PSK, it will be canceled, and later triggers the `otBorderAgentEphemeralKeyCallback`, which will then lead to unpublishing the service. 

Thoughts?

Reply:
Thanks for the comments! Will follow the suggestion to modify SetEpskcEnabled code and do verification accordingly.